lms/services/launch_verifier.py

```diff
@@ -139,11 +145,9 @@ def validate_client_key(self, client_key, request):
     def get_client_secret(self, client_key, request):
         """Retrieve the client secret associated with the client key."""
 
-        application_instance = models.ApplicationInstance.get_by_consumer_key(
-            self.db_session, client_key
-        )
-
-        if not application_instance:
-            raise ConsumerKeyLaunchVerificationError()
-
-        return application_instance.shared_secret
+        try:
+            return self.application_instance_service.get_by_consumer_key(
+                client_key
+            ).shared_secret
```

Comment:
If we used the previous `get()` method then that would read `lti_user` which is what we are creating. Thus creating a loop.